### DISCUSSION TITLE
How to delete old events linked to an asset key.

### DISCUSSION THREAD

**Discussion by marcilj** [dagster-io/dagster#18247] [Q&A] [answered]
I've change my asset key name, and since my old one still have event materialization event store in Dagster I always endup clicking on the old one instead of the new one.

Is there a way to delete an asset key that is not used anymore but have old materialization event stored in Dagster?

**Comment by benpankow** [comment, marked as answer]
Hi Jacob,

The easiest way to do this is to remove the materialization data for the old asset key, that way it will be hidden from the UI.

<img width="793" alt="Screenshot 2023-11-28 at 10 03 40 AM" src="https://github.com/dagster-io/dagster/assets/10215173/6fb36e59-bfe2-433d-a2e7-2463e00f675b">

**Reply by marcilj** [reply]
Thank you very much.

**Comment by nicchristeller** [comment]
I'm having this issue persist after I've deleted all materializations and events for an old asset key.

Dagster is now showing the asset as an "external asset", but not removing it. It shows that there are no definitions for the asset. 

How can I remove this asset from the UI?

![image](https://github.com/user-attachments/assets/75222734-1592-4e7a-b9e9-b2455025ac3f)

**Reply by matthias-Q** [reply]
I had the same issue. In the asset overview page, you can select the asset and wipe all materializations. This solve the issue for me.

**Reply by nicchristeller** [reply]
Ah, I was looking under the drop down for the asset, not the one at the top of the table. Thank you!
